inst/tests/tests.Rraw

```diff
@@ -17537,3 +17537,8 @@ vr = "Species"
 IDT[, virginca := get(vr) == "virginica"]
 ans = data.table(round = c(3, 3, 3, 2, 2, 4, 2, 4), k = c(6, 7, 8, 5, 7, 7, 6, 8), kar = structure(c(3L, 3L, 3L, 3L, 3L, 3L, 3L, 3L), .Label = c("setosa", "versicolor", "virginica"), class = "factor"), N = c(24L, 14L, 4L, 1L, 1L, 1L, 3L, 2L))
 test(2184.5, IDT[(virginca), .N, by = .(round(Sepal.Width), k = round(Sepal.Length), kar = get(vr))] , ans)
+
+# initial '' doesn't crash as.IDate, #4676;
+#   also imitates base R as from 4.0.2 patched
+y = c('', '2020-01-01')
+test(2185, as.IDate(y), rev(as.IDate(rev(y))))
```

Comment:
This test compares `as.IDate` to itself, so it could pass incorrectly. Tests should compare to a fixed value.

This was the main point of me requesting-changes on this PR.  Then the other relatively minor requests I added too while I was at it.

Reply:
thinking here was that the bug is about inconsistency in the first place -- the order within `y` matters for the output. so this test confirms the order doesn't matter.

for that reason i'll leave it, but i agree we should also test against a fixed value, so I'll add a second test.

Reply:
in your revision you still used `as.IDate` in the y value.  I changed it to a truly fixed value.